### PR TITLE
Refactor `glTFResource`

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

- Improve the rationality and ease of use of glTFResource
- Internally increase the reference counting mechanism of `template resources` and use it to release resources of glTFResource BUG

### CODE REVIEW

File: packages/loader/src/gltf/parser/GLTFMeshParser.ts [hunk 1]
```diff
@@ -326,6 +324,12 @@ export class GLTFMeshParser extends GLTFParser {
...
     }
 
-    return Promise.all(primitivePromises);
+    return Promise.all(primitivePromises).then((meshes) => {
+      for (let i = 0, n = meshes.length; i < n; i++) {
+        // @ts-ignore
+        meshes[i]._associationSuperResource(glTFResource);
+      }
+      return meshes;
+    });
```
Comment (cptbtptpbcptdtptp) [inline packages/loader/src/gltf/parser/GLTFMeshParser.ts:333]:
```

        if (mesh) {
          if (mesh instanceof ModelMesh) {
            mesh._associationSuperResource(glTFResource);
            resolve(mesh);
          } else {
            mesh.then((mesh) => {
              mesh._associationSuperResource(glTFResource);
              resolve(mesh);
            });
          }
        } else {
          const mesh = new ModelMesh(engine, meshInfo.name || i + "");

          const meshRestoreInfo = new ModelMeshRestoreInfo();
          meshRestoreInfo.mesh = mesh;
          context.contentRestorer.meshes.push(meshRestoreInfo);
          mesh._associationSuperResource(glTFResource);
          GLTFMeshParser._parseMeshFromGLTFPrimitive(
            context,
            mesh,
            meshRestoreInfo,
            meshInfo,
            gltfPrimitive,
            glTF,
            (attributeSemantic) => {
              return null;
            },
            (attributeName, shapeIndex) => {
              const shapeAccessorIdx = gltfPrimitive.targets[shapeIndex];
              const attributeAccessorIdx = shapeAccessorIdx[attributeName];
              if (attributeAccessorIdx) {
                const accessor = glTF.accessors[attributeAccessorIdx];
                return GLTFUtils.getAccessorBuffer(context, context.glTF.bufferViews, accessor);
              } else {
                return null;
              }
            },
            () => {
              const indexAccessor = glTF.accessors[gltfPrimitive.indices];
              return context.get<ArrayBuffer>(GLTFParserType.Buffer).then((buffers) => {
                return GLTFUtils.getAccessorData(glTF, indexAccessor, buffers);
              });
            },
            context.params.keepMeshData
          ).then(resolve);
        }
```

File: packages/core/src/asset/ReferResource.ts [hunk 2]
```diff
@@ -25,17 +26,54 @@ export abstract class ReferResource extends EngineObject implements IReferable {
...
+   * @param force - Whether to force the destruction, if it is false, refCount = 0 can be released successfully
+   * @returns Whether the release was successful
    */
-  override destroy(force: boolean = false): boolean {
+  override destroy(force?: boolean): boolean;
+
+  /**
+   * @internal
+   */
+  override destroy(force: boolean, isGC: boolean): boolean;
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/asset/ReferResource.ts:37]: Is it not needed here?

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]

File: packages/loader/src/gltf/parser/GLTFMeshParser.ts (hunk 1 above)
Comment (GuoLei1990) [reply packages/loader/src/gltf/parser/GLTFMeshParser.ts:333]: Done

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**